proto/proto-backplane-grpc/src/main/proto/deephaven/proto/object.proto

```diff
@@ -12,7 +12,67 @@ option go_package = "github.com/deephaven/deephaven-core/go/internal/proto/objec
 import "deephaven/proto/ticket.proto";
 
 service ObjectService {
-    rpc FetchObject(FetchObjectRequest) returns (FetchObjectResponse) {}
+
+    /*
+     * Fetches a server-side object as a binary payload and assorted other tickets pointing at
+     * other server-side objects that may need to be read to properly use this payload. The binary
+     * format is implementation specific, but the implementation should be specified by the "type"
+     * identifier in the typed ticket.
+     *
+     * Deprecated in favor of MessageStream, which is able to handle the same content.
```

Comment:
I'm _assuming_ that moving to MessageStream would entail:

1. client sends ConnectRequest equivalent of FetchObjectRequest to server. (Is it correct for client to call onComplete() after onNext() since it knows it won't be sending anymore messages? I see note later about half-closed stuff...)
2. server sends Data payload equivalent of FetchObjectResponse. (I'm assuming it is always correct for the server to call onComplete() after onNext() in this case.)

I could imagine the server / client implementations to have enough knowledge about how the plugin works to error out in the case where either the server or client doesn't adhere to the one-shot nature.

Reply:
1. Technically yes, the client could (perhaps even "should", but I can't see a compelling case for that) half-close after sending the message, but as implemented we don't do that (and don't _have_ to do it either). The issue here is that the client API, as written (more on that in a moment) doesn't know the difference between bidi and fetch widgets. 
2. If the server knows that it is dealing with a "fetch" widget, then it knows to close after sending, yes.

As the API is designed today, the wire format doesn't know the difference between these types of widgets at all. The server implementation does, as a nicety to let old widgets keep working and also as an admission that fetch-only widgets are a very reasonable use case by themselves, which don't need long lived streams. Technically we could make fetch-only widgets be implemented as streaming ones, and just close right away... but combining that extra work with the existing legacy widgets, and I'm inclined to support both versions going forwards.

If we were willing to break the API entirely, I'd change the ObjectType API further so that the an implementation picks which interface it will use, "fetchable" vs "bidirectional". Do we agree that the benefit probably isn't worth it?

Client API: The existing api of a Js dh.Widget object is that it is fetched asychronously, and then provides two accessors - one for the payload, and one for the array of nested objects (which can in turn be fetched asynchronously). The modifications for this patch only add an event that dh.Widget can emit, which results in _subsequent_ responses being provided to the caller this way - the first response must arrive on the client so that we know the fetch worked at all (or, we need another response type, to indicate "yep, it existed, first payload will arrive shortly", see the notes on sending a response right away). This keeps perfect compatibility with existing plugins, and only new bidi ones have to deal with adding this event handler. Failing to treat a streaming plugin as streaming results in lost data, leaked objects on the server, leaked stream on the client, but isn't necessarily fatal... Perhaps it could/should be?